modules/tensor_mechanics/src/neml2/userobjects/CauchyStressFromNEML2UO.C

```diff
@@ -45,11 +57,49 @@ CauchyStressFromNEML2UO::CauchyStressFromNEML2UO(const InputParameters & params)
 
 CauchyStressFromNEML2UO::CauchyStressFromNEML2UO(const InputParameters & params)
   : NEML2SolidMechanicsInterface<CauchyStressFromNEML2UOParent>(
-        params, "mechanical_strain", "temperature")
+        params, "mechanical_strain", "temperature"),
+    _parameter_derivatives(getParam<std::vector<std::string>>("parameter_derivatives")),
+    _require_parameter_derivatives(_parameter_derivatives.size() ? true : false),
+    _reset_parameter_names(getParam<std::vector<std::string>>("reset_parameter_names")),
+    _reset_parameters(_reset_parameter_names.size() ? true : false),
+    _material_param(_reset_parameters
+                        ? &getMaterialProperty<Real>(getParam<std::string>("param_material_prop"))
+                        : nullptr),
+    _material_param_uo(_reset_parameters ? &getUserObject<BatchScalarProperty>("material_param_uo")
+                                         : nullptr)
 {
   validateModel();
 }
 
+void
+CauchyStressFromNEML2UO::preCompute()
+{
+  // Set requires_grad_() for each parameter if we request stress derivatives w.r.t that parameter
+  if (_require_parameter_derivatives)
+  {
+    for (auto param_name : _parameter_derivatives)
+    {
+      auto model_param = model().named_parameters(true)[param_name];
+      model_param.requires_grad_();
+    }
+  }
+
+  // Set the parameter value using batch material from MOOSE
+  if (_reset_parameters)
+  {
+    // TODO: extend to multiple parameters
+    if (!_material_param_uo->outputReady())
+      mooseError("Batch material parameter is not ready for output");
+
+    auto model_param = model().named_parameters(true)[_reset_parameter_names[0]];
+    auto param_values = NEML2Utils::toNEML2Batched(_material_param_uo->getOutputData());
+
+    model_param = model_param.batch_expand_copy(param_values.batch_sizes());
+
+    model_param.copy_(param_values);
+  }
+}
```

Comment:
@hugary1995 Thanks for your recent updates in neml2 which makes the `batch_expand_copy` to expand the scalar to proper batch size without needing to hard-code it in the input file. 

However, I still find that the parameter is not correctly set and/or consumed in neml2 material model. Therefore, would you mind helping me check if this is the correct way of setting material properties? Or am I not calling `precompute()` in the proper location?

Reply:
> Can this wait a few days as I wrap up the current (giant) NEML2 refactor?

No prob at all - thanks for considering this in your current neml2 refactor.